validator.go

```diff
@@ -0,0 +1,201 @@
+package jwt
+
+import (
+	"crypto/subtle"
+	"fmt"
+	"time"
+)
+
+// Validator is the core of the new Validation API. It is
+type Validator struct {
+	// leeway is an optional leeway that can be provided to account for clock skew.
+	leeway time.Duration
+
+	// timeFunc is used to supply the current time that is needed for
+	// validation. If unspecified, this defaults to time.Now.
+	timeFunc func() time.Time
+
+	// verifyIat specifies whether the iat (Issued At) claim will be verified.
+	// According to https://www.rfc-editor.org/rfc/rfc7519#section-4.1.6 this
+	// only specifies the age of the token, but no validation check is
+	// necessary. However, if wanted, it can be checked if the iat is
+	// unrealistic, i.e., in the future.
+	verifyIat bool
+
+	// expectedAud contains the audiences this token expects. Supplying an empty
+	// string will disable aud checking.
+	expectedAud string
+}
+
+type customValidationType interface {
+	CustomValidation() error
+}
```

Comment:
It seems to me this new interface, `customValidationType`, could be implemented by a package user to inject their own custom validation of claims. If this is the case, I think this interface should be exported and this feature should be documented, perhaps in the `README.md`.

May I suggest renaming it to `CustomValidation` and its method to `Validate`?

Reply:
> It seems to me this new interface, `customValidationType`, could be implemented by a package user to inject their own custom validation of claims. If this is the case, I think this interface should be exported and this feature should be documented, perhaps in the `README.md`.

Yes that is exactly the use case! Including this in the README makes perfect sense.

> 
> May I suggest renaming it to `CustomValidation` and its method to `Validate`?

The way it works is that it checks *on top* of the checks performed by jwt and therefore it wanted something that sounds like an additional verification of custom claims, rather than something that sounds like it can override basic validation.